### PR TITLE
Administration (ILIAS 9): Feature: common administrative node »Legal Regulations«

### PR DESCRIPTION
**Pull request by fhelfer**
Feature Wiki: https://docu.ilias.de/goto_docu_wiki_wpage_7564_1357.html

### CODE REVIEW

File: Services/Imprint/classes/Setup/ilImprintUpdateAgent.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Setup;
+use ILIAS\Refinery;
+use ILIAS\Setup\Objective\ObjectiveWithPreconditions;
+
+class ilImprintUpdateAgent implements Setup\Agent
```
Comment (lscharmer) [inline Services/Imprint/classes/Setup/ilImprintUpdateAgent.php]: Please put htis file into a namespace (and remove the `il` prefix).

File: Services/Imprint/classes/class.ilObjLegalNoticeGUI.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+use ILIAS\UI\Component\Component;
+
+/**
+ * @author            Michael Jansen <mjansen@databay.de>
+ * @ilCtrl_Calls      ilObjLegalNoticeGUI: ilPermissionGUI, ilImprintGUI
+ * @ilCtrl_isCalledBy ilObjLegalNoticeGUI: ilAdministrationGUI
+ */
+class ilObjLegalNoticeGUI extends ilObject2GUI
+{
+    private ilImprintGUI $impressum;
```
Comment (lscharmer) [inline Services/Imprint/classes/class.ilObjLegalNoticeGUI.php]: Please translate this property into english.

File: src/UI/Component/Symbol/Icon/Standard.php [hunk 3]
```diff
@@ -108,6 +108,8 @@ interface Standard extends Icon
     public const RTST = 'rtst';	//Remote Test Object
     public const ECSS = 'ecss';	//ECS Administration
     public const TOS = 'tos';	//Terms of Service
+
+    public const IMPR = 'impr';	//Impressum
```
Comment (lscharmer) [inline src/UI/Component/Symbol/Icon/Standard.php]: Please write your comment in english.

**Review by lscharmer — CHANGES_REQUESTED** [review]

**Review by lscharmer — APPROVED** [review]

**Comment by fhelfer** [comment]
Because of the dependency on FR: https://docu.ilias.de/goto_docu_wiki_wpage_7451_1357.html
There will be a followup PR for moving the DataProtection to Legal-Documents after PR https://github.com/ILIAS-eLearning/ILIAS/pull/6451 has been merged.

For the following files i will need a review of the mentioned maintainers.
@alex40724 for `Modules/SystemFolder`, `Services/Administration`
@Amstutz @klees for `src/UI`

Best, @fhelfer

**Comment by mjansenDatabay** [comment]
@fhelfer IMO we should create a separate PR for the `src/UI` changes.

**Review by alex40724 — APPROVED** [review]
Thanks!

File: Services/Imprint/classes/class.ilObjLegalNotice.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+class ilObjLegalNotice extends ilObject2
+{
+    protected ilSetting $settings;
```
Comment (mjansenDatabay) [inline Services/Imprint/classes/class.ilObjLegalNotice.php]: Do we need this property? If not, please remove it.

File: Services/Imprint/classes/class.ilObjLegalNoticeGUI.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        $cmd = $this->ctrl->getCmd();
+
+
+        switch (strtolower($nextClass)) {
+            case strtolower(ilPermissionGUI::class):
+                $perm_gui = new ilPermissionGUI($this);
+                $this->ctrl->forwardCommand($perm_gui);
+                break;
+            default:
+                if (in_array($cmd, ['', 'view', 'edit', null], true) || !method_exists($this, $cmd)) {
```
Comment (mjansenDatabay) [inline Services/Imprint/classes/class.ilObjLegalNoticeGUI.php]: Why do we need `|| !method_exists($this, $cmd)`?

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
Thx @fhelfer ,

- [ ] Please check my inline comments
- [ ] Please create a separate PR for the `src/UI` changes.

Best regards,
Michael

File: Services/Imprint/classes/class.ilObjLegalNoticeGUI.php (hunk 5 above)
Comment (fhelfer) [reply Services/Imprint/classes/class.ilObjLegalNoticeGUI.php]: for commands like invokeServer & finishEditing. Not sure if there were more

Comment (fhelfer) [reply Services/Imprint/classes/class.ilObjLegalNoticeGUI.php]: https://github.com/ILIAS-eLearning/ILIAS/pull/5791/commits/e19bb712dceb890a49a15dbd2c47c7e9d97c2166#diff-3dd562884774a8008bdf2403959ed9dcfdabed7a50eebf62b5823333c8ff4abaR58-R69
this seems better

**Comment by fhelfer** [comment]
> * [x]  Please check my inline comments
> * [x]  Please create a separate PR for the `src/UI` changes.
> 
> Best regards, Michael

Seperate PR has been created: https://github.com/ILIAS-eLearning/ILIAS/pull/6457

Done
Best, @fhelfer

**Comment by oliversamoila** [comment]
The final icons for "Legal Regulations" as well as the admin nodes below will follow in the beta. Unfortunately, they are not yet available.

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**